apps/cow-tools/src/app/milkman/[network]/order/new/page.tsx

```diff
@@ -0,0 +1,426 @@
+    return <WalletNotConnected />;
+  }
+
+  if (isConnecting || isReconnecting) {
+    return <Spinner />;
+  }
+
+  const network = getNetwork(chain?.name);
+
+  if (network !== params.network) {
+    return (
+      <div className="flex h-full w-full flex-col items-center rounded-3xl px-12 py-16 md:py-20">
+        <div className="text-center text-3xl text-amber9">
+          You are on the wrong network
+        </div>
+        <div className="text-xl text-white">
+          Please change to {params.network}
+        </div>
+      </div>
+    );
+  }
+
+  return (
+    <>
+      <TransactionCard
+        userAddress={addressLower as Address}
+        chainId={chain?.id}
+      />
+    </>
+  );
+}
+
+function TransactionCard({
+  userAddress,
+  chainId,
+}: {
+  userAddress: Address;
+  chainId?: NetworkChainId;
+  walletAmount?: string;
+  walletAmountBigNumber?: bigint;
+}) {
+  const { transactionStatus, setTransactionStatus } = useOrder();
+  const network = networkFor(chainId);
+
+  const form = useForm();
+  const { register, setValue, watch, clearErrors } = form;
+
+  useEffect(() => {
+    register("receiverAddress");
+    setValue("tokenAddress", "0x768788EB28d25C351E502cd7c5882C63C0876237");
+  }, []);
+
+  const formData = watch();
+
+  useEffect(() => {
+    if (formData.isValidFromNeeded === false) {
+      setValue("validFrom", null);
+      clearErrors("validFrom");
+    }
+  }, [formData]);
+
+  function handleOnSubmit(data: FieldValues) {
+    // eslint-disable-next-line no-console
+    console.log(data);
+    alert("submited");
+  }
+
+  function handleBack() {
+    const currentStage = stages.find(
+      (stage) => stage.name === transactionStatus,
+    );
+    setTransactionStatus(currentStage?.previousStage ?? transactionStatus);
+  }
+
+  function handleContinue() {
+    const currentStage = stages.find(
+      (stage) => stage.name === transactionStatus,
+    );
+    setTransactionStatus(currentStage?.nextStage ?? transactionStatus);
+  }
+
+  const FORM_CONTENTS: { [key in TransactionStatus]?: JSX.Element } = {
+    [TransactionStatus.DRAFT_SELECT_TOKENS]: (
+      <FormSelectToken form={form} userAddress={userAddress} />
+    ),
+    [TransactionStatus.DRAFT_SELECT_PRICE_CHECKER]: (
+      <FormSelectPriceChecker form={form} />
+    ),
+    [TransactionStatus.DRAFT_RESUME]: (
+      <div className="flex flex-col gap-y-6 p-9">
+        <OrderResume data={formData} handleBack={handleBack} />
+      </div>
+    ),
+  };
+
+  return (
+    <div className="flex h-full items-center justify-center w-full mt-20">
+      <Form
+        {...form}
+        onSubmit={handleOnSubmit}
+        className="my-4 flex h-fit w-fit flex-col  rounded-lg border border-slate7 bg-blue3 text-white"
+      >
+        <div className="divide-y divide-slate7">
+          <FormHeader
+            transactionStatus={transactionStatus}
+            network={network}
+            onClick={handleBack}
```

Comment:
imo it isn't intuitive why we need the network here. It is just to handle the back on the first stage, right? If so, I would try to generalize this to just one handleBack function that already handles both cases.

Reply:
network is needed to create href for the LinkComponent, onClick shouldn't handle internal route changes